htdocs/core/class/html.form.class.php

```diff
@@ -3334,7 +3334,7 @@ protected function constructProductListOption(&$objp, &$opt, &$optJson, $price_l
 
 		if (isModEnabled('stock') && isset($objp->stock) && ($objp->fk_product_type == Product::TYPE_PRODUCT || getDolGlobalString('STOCK_SUPPORTS_SERVICES'))) {
 			if ($user->hasRight('stock', 'lire')) {
-				$opt .= ' - ' . $langs->trans("Stock") . ': ' . price(price2num($objp->stock, 'MS'));
+				$opt .= ' - ' . $langs->trans("Stock") . ': ' . price2num($objp->stock, 'MS');
```

Comment:
Can you try 
`price(price2num($objp->stock, 'MS'), 0, $langs, 0, 0)`
So the format of number will be in the user language format instead of php international format.

Reply:
Your suggestion works perfectly:
- respects the setting of MAIN_MAX_DECIMALS_STOCK
- never prints more decimals than required, i.e. 0 instead of 0.00

![image](https://github.com/user-attachments/assets/c0fff932-7f11-40d0-aa14-aee6d0964b6d)